src/main/java/deku/InputParser.java

```diff
@@ -0,0 +1,158 @@
+package deku;
+
+import java.time.DateTimeException;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+
+import deku.task.Deadline;
+import deku.task.Event;
+import deku.task.ToDo;
+
+/**
+ * Parses input from user to bot understandable structure
+ */
+public class InputParser {
+    private LocalDate date;
+    private LocalTime time;
+
+    /**
+     * Parses input from user to bot understandable structure
+     *
+     * @param task Space separated List of instruction after the user command
+     * @return String of friendly and readable output
+     */
+    public String parseTask(List<String> task) {
+        String date = "";
+        String misc = "";
+        String time = "";
+        String output = "";
+        boolean isDateTime = false;
+        int index = 0;
+        while (index < task.size()) {
+            String top = task.get(index);
+            if (top.charAt(0) == '/') {
+                isDateTime = true;
+                output += "(" + top.substring(1) + ": ";
+                index++;
+                continue;
+            }
+            if (isDateTime) {
+                if (date.equals("")) {
+                    date = top;
+                } else {
+                    time = top;
+                }
+                misc += top + " ";
+            } else {
+                output += top + " ";
+            }
+            index++;
+        }
+        if (!misc.equals("")) {
+            misc = misc.substring(0, misc.length() - 1);
+        }
+        if (isDateTime) {
+            parseDate(date);
+            if (this.date == null) {
+                return output + misc + ")";
+            }
+            output += this.date.format(DateTimeFormatter.ofPattern("d MMM yyyy"));
+            if (time.equals("")) {
+                return output + ")";
+            }
+            parseTime(time);
+            if (this.time == null) {
+                return output + " " + time + ")";
+            }
+            return output
+                   + " "
+                   + this.time.format(DateTimeFormatter.ofPattern("HH:mm"))
+                   + ")";
+        }
+        return output.substring(0, output.length() - 1);
+    }

```

Comment:
> Be wary when a method is longer than the computer screen, and take corrective action when it goes beyond 30 LOC (lines of code). The bigger the haystack, the harder it is to find a needle.

Maybe you can parse your method into several methods instead. This will help improve readability of the code. 👍